IviDriverCore/1.0/Spec/IviDriverCore.md

```diff
@@ -0,0 +1,770 @@
+
+### Extent of Instrument Functionality Covered by IVI Drivers
+
+IVI drivers for instruments that have an ASCII command set such as SCPI shall implement the full functionality of the instrument available via commands and queries with a few exceptions. Some commands and queries are not suitable for an instrument driver or could even break driver or instrument functionality if exposed to the user. For example, an IVI driver for a SCPI-based instrument might not implement SCPI commands from the following nodes:
+
+- `DIAGnostic`
+
+- `FORMat` (may be used internally but not exposed to users)
+
+- `SYSTem:COMMunicate`
+
+- Service or Factory Calibration functionality
+
+- Undocumented SCPI (factory use only)
+
+- Other features not normally accessed through the programmatic interface, for example:
+
+  - `DISPlay`
+
+  - `HARDCopy`
+
+  - `MEMory:STATe`
+
+  - `CURSor`
+
+> **Observation:**
+> > IVI driver users can send any commands to a message-based instrument using the driver's Direct I/O functions.
+
+### Direct I/O Requirement
+
+IVI Drivers for instruments that have a supported ASCII command set, such as SCPI, shall also provide an API for sending messages to and from the instrument over the ASCII command channel. The details of the API are specified in [Direct IO API](#direct-io).
+
+### Query Instrument Status on API Call
+
+If the instrument can be queried for its status it shall provide a mechanism to automatically query the instrument status on every API call. This mechanism is controlled with [Query Instrument Status Enabled](#query-instrument-status-enabled).
+
+## Required Driver APIs
+
+This section gives a functional description of the members of the API required for an IVI Core driver.
+
+The general function of each member of the API is described here. However, this specification does not generally indicate what form a particular member should take.
+
+For example:
+
+- Initialization may be implemented as a constructor in .NET, but might be implemented as a method in another language.
+
+- Supported Instrument Models is implemented as a method called `GetSupportedInstrumentModels()` in .NET, but might be implemented as a property in another language.
+
+Refer to language-specific specifications for implementation details.
+
+### Initialization (Construction)
+
+Depending on the language, IVI Core Drivers may be initialized in a constructor or in a method. This process is called *initialization* in this section.
+
+The following parameters are defined for the *initialization* operation:
+
+- ***Resource Name*:**
+  An instrument specific string that identifies the address of the instrument, such as a VISA resource descriptor string.
+
+- ***ID Query*:**
+  Specifies if the identity of the instrument should be verified during *initialization*.
+
+- ***Reset*:**
+  Specifies if the instrument should be reset during *initialization*.
+
+- ***Simulation*:**
```

Comment:
For my own info since I was only recently included in the IVI discussions:

Was option string dropped as an initialization parameter requirement? In our existing drivers this is definitely an important part of our initialization setup and what allows us to flexibly add new initialization and simulation options as needs arise.

I believe based off the wording here is that anything not on this list is simply optional now, which means there's no worries about IVI compliance if we already utilize option string, but I could see this causing future drivers trying to be IVI compliant to have considerably different initialization signatures. I imagine this was a conscious decision, so I don't personally believe it's something that needs to be resolved, but I wanted to see what the full reasoning and decision was.

I'll be OOO until January 16th after today so I won't be able to respond to any discussions that might happen from this.

Thanks!